drivers/timer/legacy_api.h

```diff
@@ -0,0 +1,69 @@
+#ifndef ZEPHYR_LEGACY_SET_TIME_H__
+#define ZEPHYR_LEGACY_SET_TIME_H__
+
+/* Stub implementation of z_clock_set_timeout() and z_clock_elapsed()
+ * in terms of the original APIs.  Used by older timer drivers.
+ * Should be replaced.
+ *
+ * Yes, this "header" includes function definitions and must be
```

Comment:
> Maybe you want to declare the function static to save this note?
>> It's a public function. Basically the legacy_api.h "header" is a stub of C code that the driver code can include to implement the new API in terms of the old one. It's grown another wart or two in the next pull request too.

Yes, I got that :). 
Even though static is normally understood as “private”, I was going for the true meaning of static (scope limited to the translation unit)
(This is a very minor thing anyhow)
So, my point was that, if you add static to ```z_clock_set_timeout  & z_clock_elapsed``` below, then the header can be “safely” included multiple times. Just each time it is included, a new copy of that function is created, without name-clashes with wherever it would be included again.

It does not really matter as there would only be 1 driver, and therefore the header should be included only once.
But it would save you from needing to warn people to not need to include the header more than once.

Reply:
Yes, but those APIs are exported functions from the timer driver that need to be called by the kernel.  They must be non-static.  But they're also declared in a "header" due to the crazy shim architecture, thus the warning.

Again, it's all sort of academic anyway.  The drivers need to be reworked to drop the whole scheme and this header will go away.